### ISSUE TITLE
JSON serialization of  fails on bool and NoneType

### ISSUE THREAD

**Issue by davidt0x** [ModECI/MDF#49]
See failing tests on: cf826e28b6676847aa1835d7f735558046a2ebaa.

`NoneType` and `bool` values are causing JSON serialization errors.  I think this bug is somewhere in neuromllite @pgleeson. Python's json serialization supports this, are we not using it?

```
import json
json.dumps({'test1': None, 'test2': False, 'test3': True})
Out[41]: '{"test1": null, "test2": false, "test3": true}'
```

**Comment by pgleeson** [comment]
@davidt0x I can certainly try to support bool values (just haven't considered them yet) but do you really need None? It may break other things down the line, i.e. when converting the serialised values to classes/objects.

**Comment by pgleeson** [comment]
@davidt0x bool is now supported in neuromllite (a pull/install from master required), see https://github.com/NeuroML/NeuroMLlite/pull/17/files. Let me know if you really need None/null..

**Comment by davidt0x** [comment]
Hey, thanks, sorry just noticed your question about `None`. `None` is useful because a decent amount of torch functions use `None` to specify defaults for optional arguments it seems. This is true for Python in general really, `typing.Optional` is really the same as `typing.Union[None, ...]`. I was trying to add a ton of torch functions to the supported functions API and it would make it easier to do this

**Comment by pgleeson** [comment]
Ok, but do you really want to specify None as the type of a field in the definition, or do you want to have the option to put None in when a variable of a particular type is unknown? See this example: https://github.com/NeuroML/NeuroMLlite/blob/5920c4e894486f5c0674ce5b78ded90aa4d534bb/neuromllite/test/test_base.py#L109-L112 where None can be set for a field when a bool or a string is required and it can write/reload the objects correctly.

**Comment by davidt0x** [comment]
Sorry I am a bit confused with the question. I don't think I want to be able to specify None as the type, but as the actual value. I want to do what I am trying to do in this failing test:
[test_model.test_none_json](https://github.com/ModECI/MDF/blob/17c5a279d9401b6291f3f57f403221b792931738/tests/test_model.py#L65).

`Node(id="test_node", parameters={"test": None}).to_json()`

Does this make sense?

**Comment by pgleeson** [comment]
Ok, it was treating it slightly differently if it was in a dict. The latest change on master should work now. 

https://github.com/NeuroML/NeuroMLlite/blob/master/neuromllite/test/test_base.py#L96

**Comment by davidt0x** [comment]
Hey @pgleeson,

I am going to reopen this issue, This fix seems to be up on neuromllite master but not in latest PyPI package. Can you check?

**Comment by pgleeson** [comment]
@davidt0x Good point. I've released v0.3.2 with all latest changes to pypi.

**Comment by pgleeson** [comment]
Closing as fixed in development
